### DISCUSSION TITLE
3D-objects appear without texture/mesh on iOS 15+

### DISCUSSION THREAD

**Discussion by bankey1443** [google/model-viewer#2923] [Q&A]
I wasn't sure if I should open up an issue for this or not, but on the latest iOS version 15+ it seems like the model-viewer sometimes does not load in textures. While this seems to only affect a small subset of our 3d-objects, it seems pretty consistent to replicate on my iPhone. On first load it usually displays the 3d-model without any color, or partially covered (see first three screenshots), and if you refresh and try again, the page seems to crash or refresh (see 4th screenshot). AR works as intended.

![noSkin](https://user-images.githubusercontent.com/10816600/138892499-eace05d2-d60d-4bad-a76f-ad53add7a285.jpeg)
![halfSkin](https://user-images.githubusercontent.com/10816600/138892525-11bdab89-8683-4897-b171-2dfd5903fb8f.png)
![IMG_2719](https://user-images.githubusercontent.com/10816600/138906987-a40271d9-50b0-4c27-9d52-bad5e553c726.PNG)
![IMG_2718](https://user-images.githubusercontent.com/10816600/138892931-d3ff1386-7e00-4273-89d1-d601b049381c.PNG)

The following is a screenshot of what the object should look like. Most of our products are black/gray/steel colored, and the two that consistently do not work are colored bright red/yellow, although that might just be a coincidence:

<img width="331" alt="fullSkin" src="https://user-images.githubusercontent.com/10816600/138907150-48726737-41c3-44aa-90ba-d54cdc7193bd.png">

We have tested this on the following devices, using model-viewer v1.4.1:
Android - working
iPhone XR/Safari/iOS 13 - working
iPhone 7/Safari/iOS 15 - not working
iPhone XR/Safari/iOS 15.0.2 - not working

Here is a live demo where, on my iPhone, it either works, displays either black/gray (without textures), refreshes the page, or crashes the page seemingly randomly when I try to test it by refreshing the page constantly: https://crawling-bronzed-dash.glitch.me/

If more examples are required, let me know.

**Comment by elalish** [comment]
Yes, we've been seeing this occasionally  as well, though it's been hard to repro. Thanks for the repro  link;  if it's more  consistent that will help a lot. This appears to be a Safari WebGL bug, and I'm not sure if there's a  workaround  for it. Since your link is MV 1.9 and  you were also  testing MV 1.4.1, I'd say this is not  a MV regression.

**Reply by bankey1443** [reply]
Thanks for the reply. I assume from your comment that this can't (or won't) be fixed by an update to MV, which means that we will need to wait for an update for WebGL, correct?

**Reply by elalish** [reply]
And update to Safari (WebKit), yes. If you wouldn't mind filing a bug on WebKit [here](https://bugs.webkit.org/), that's our best shot at getting it fixed.

**Reply by bankey1443** [reply]
Filed a bug on WebKit https://bugs.webkit.org/show_bug.cgi?id=232378
